velox/exec/SetAccumulator.h

```diff
@@ -106,10 +153,50 @@ struct SetAccumulator {
                                  : uniqueValues.size();
   }
 
+  void serializeNullIndex(char* buffer) {
+    auto nullIndexValue =
+        nullIndex.has_value() ? nullIndex.value() : kNoNullIndex;
+    memcpy(buffer, &nullIndexValue, kVectorSizeT);
+  }
+
+  /// Extracts in result[index] a serialized VARBINARY for the Set Values.
+  /// This is used for the spill of this accumulator.
+  void serialize(const VectorPtr& result, vector_size_t index) {
+    // The serialized value is the nullOffset (kNoNullIndex if no null is
+    // present) followed by the unique values in order of their indices.
+    // The null position is skipped when serializing the values.
+    size_t totalBytes = kVectorSizeT + kValueSizeT * uniqueValues.size();
+
+    auto* flatResult = result->as<FlatVector<StringView>>();
+    auto* rawBuffer = flatResult->getRawStringBufferWithSpace(totalBytes, true);
+
+    serializeNullIndex(rawBuffer);
+
+    // The null position is skipped when serializing values, so setting an out
+    // of bound value for no null position.
+    const auto nullPosition =
+        nullIndex.has_value() ? nullIndex : uniqueValues.size();
+    // Compute offset in serialization buffer for a value at 'index' position.
+    auto offset = [&](vector_size_t index) {
```

Comment:
this lambda is used only once; it might be easier to read if the code was inlined